django/core/management/templates.py

```diff
@@ -77,6 +78,10 @@ def handle(self, app_or_project, name, target=None, **options):
             if app_or_project == 'app':
                 self.validate_name(os.path.basename(target), 'directory')
             top_dir = os.path.abspath(path.expanduser(target))
+            app_dir = top_dir.split('/')
+            working_dir = os.getcwd().split('/')
+            app_python_dir = [item for item in app_dir if item not in working_dir]
+            app_python_path = '.'.join(map(str, app_python_dir))
```

Comment:
I think we should use `relpath()` but only if it is below the working directory, e.g.
```diff
diff --git a/django/core/management/templates.py b/django/core/management/templates.py
index c7252a5ad2..db417443b2 100644
--- a/django/core/management/templates.py
+++ b/django/core/management/templates.py
@@ -65,6 +65,7 @@ class TemplateCommand(BaseCommand):
         self.validate_name(name)
 
         # if some directory is given, make sure it's nicely expanded
+        app_python_path = name
         if target is None:
             top_dir = path.join(os.getcwd(), name)
             try:
@@ -77,6 +78,11 @@ class TemplateCommand(BaseCommand):
             if app_or_project == 'app':
                 self.validate_name(os.path.basename(target), 'directory')
             top_dir = os.path.abspath(path.expanduser(target))
+            # Use a relative path if it's below the current working
+            # directory, or an app name otherwise.
+            rel_path = os.path.relpath(top_dir)
+            if not rel_path.startswith('..'):
+                app_python_path = rel_path.replace('\\', '/').replace('/', '.')
             if not os.path.exists(top_dir):
                 raise CommandError("Destination directory '%s' does not "
                                    "exist, please create it first." % top_dir)
@@ -101,6 +107,7 @@ class TemplateCommand(BaseCommand):
 
         context = Context({
             **options,
+            **({'app_python_path': app_python_path} if app_or_project == 'app' else {}),
             base_name: name,
             base_directory: top_dir,
             camel_case_name: camel_case_value,
```